src/authentication/controllers/fetchSlackUser.ts

```diff
@@ -0,0 +1,23 @@
+import { Request, Response, NextFunction } from "express";
+import { fetchSlackUser } from "../services/fetchSlackUser";
+import { CustomError } from "../../errors/customErrorType";
+import { slackUserQuerySchema } from "../models/slackUserSchema";
+
+export const getSlackUserDetail = async (req: Request, res: Response, next: NextFunction) => {
+        const parsedQuery = slackUserQuerySchema.parse(req.query);
+        if (Object.keys(parsedQuery).length === 0){
+            return next(new CustomError("Bad Request", 400))
+        }
+        const { userId, userEmail: email, userName } = parsedQuery;
+        try{
+            const user = await fetchSlackUser({ userId, email, userName });
+
+        if (!user) {
+            return next(new CustomError("User not found", 404));
+        }
+        res.json(user);
+    }catch(error: any){
```

Comment:
As you have created your own Error Type, please use that, use of `any` is not good.

Reply:
should I do something like this? 
````
catch(error: unknown){
        const message = error instanceof Error? error.message: "An unexpected error occured";
        return next(new CustomError(`Something went wrong: ${message}`, 500))
    };
````

Reply:
This is what i mean:

`error: CustomError`

Reply:
I got this error when doing so: 
<img width="1001" alt="image" src="https://github.com/user-attachments/assets/f5e56395-b8ad-41b3-942d-316b76b828f3">

Reply:
Ah, so you have created that as a a class but not as a type. Leave that for now.